service/worker/pernamespaceworker_test.go

```diff
@@ -202,9 +202,11 @@ func (s *perNsWorkerManagerSuite) TestMultiplicity() {
 		Enabled: false,
 	}).AnyTimes()
 
-	s.serviceResolver.EXPECT().Lookup("ns3/0").Return(membership.NewHostInfoFromAddress("self"), nil)
-	s.serviceResolver.EXPECT().Lookup("ns3/1").Return(membership.NewHostInfoFromAddress("other"), nil)
-	s.serviceResolver.EXPECT().Lookup("ns3/2").Return(membership.NewHostInfoFromAddress("self"), nil)
+	s.serviceResolver.EXPECT().LookupN("ns3", 3).Return([]membership.HostInfo{
```

Comment:
this is not a possible return value from serviceResolver, as currently implemented

Reply:
@dnr could you elaborate?

Reply:
I mean in the current state of the PR, serviceResolver never returns duplicates. So the test is fine for what it is, but the multiplicity stuff won't actually happen in practice and we want it to. That's the danger with using mocks of course. If we wanted to do better: I don't think we want to fire up ringpop in a unit test, but maybe in a functional test.